### ISSUE TITLE
[bitnami/postgresql] v8.9.2 suddenly no longer has bug fix

### ISSUE THREAD

**Issue by middagj** [bitnami/charts#2449]
**Which chart**:
bitnami/postgresql v8.9.2

**Describe the bug**
In #2392 I fixed a bug regarding pod labels. This fix got released as version 8.9.2 which worked until today. When I compare the newest version with the one in my cache I see indeed differences.

**To Reproduce**
Install `bitnami/postgresql` version v8.9.2 with pod labels deleting cache
```shell
rm $XDG_CACHE_HOME/helm/repository/postgresql-8.9.2.tgz
helm install --version 8.9.2 -f test.yaml test bitnami/postgresql
```
with `test.yaml`:
```yaml
master:
  podLabels:
    a: b
```

**Expected behavior**
I expect that deploying same version uses the same Helm chart and that a fix does not suddenly disappears.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.0", GitCommit:"e11b7ce3b12db2941e90399e874513fbd24bcb71", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.10", GitCommit:"7a578febe155a7366767abce40d8a16795a96371", GitTreeState:"clean", BuildDate:"2019-05-01T04:05:01Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
I compared my old cached 8.9.2 with the new one. This is the difference
```patch
diff -ur a/templates/_helpers.tpl b/templates/_helpers.tpl
--- a/templates/_helpers.tpl	2020-04-23 09:55:01.000000000 +0200
+++ b/templates/_helpers.tpl	2020-04-27 11:41:58.000000000 +0200
@@ -254,6 +254,15 @@
 {{- end -}}
 
 {{/*
+Return true if a configmap should be mounted with PostgreSQL configuration
+*/}}
+{{- define "postgresql.mountConfigurationCM" -}}
+{{- if or (.Files.Glob "files/postgresql.conf") (.Files.Glob "files/pg_hba.conf") .Values.postgresqlConfiguration .Values.pgHbaConfiguration .Values.configurationConfigMap }}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
 Get the initialization scripts ConfigMap name.
 */}}
 {{- define "postgresql.initdbScriptsCM" -}}
diff -ur a/templates/statefulset-slaves.yaml b/templates/statefulset-slaves.yaml
--- a/templates/statefulset-slaves.yaml	2020-04-23 09:55:01.000000000 +0200
+++ b/templates/statefulset-slaves.yaml	2020-04-27 11:41:58.000000000 +0200
@@ -82,10 +82,10 @@
             - /bin/sh
             - -cx
             - |
-              {{ if .Values.persistence.enabled }}
-              mkdir -p {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
-              chmod 700 {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
-              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | \
+              {{- if .Values.persistence.enabled }}
+              mkdir -p {{ .Values.persistence.mountPath }}/data {{- if (include "postgresql.mountConfigurationCM" .) }} {{ .Values.persistence.mountPath }}/conf {{- end }}
+              chmod 700 {{ .Values.persistence.mountPath }}/data {{- if (include "postgresql.mountConfigurationCM" .) }} {{ .Values.persistence.mountPath }}/conf {{- end }}
+              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 {{- if not (include "postgresql.mountConfigurationCM" .) }} -not -name "conf" {{- end }} -not -name ".snapshot" -not -name "lost+found" | \
               {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
                 xargs chown -R `id -u`:`id -G | cut -d " " -f2`
               {{- else }}
diff -ur a/templates/statefulset.yaml b/templates/statefulset.yaml
--- a/templates/statefulset.yaml	2020-04-23 09:55:01.000000000 +0200
+++ b/templates/statefulset.yaml	2020-04-27 11:41:58.000000000 +0200
@@ -40,7 +40,7 @@
         heritage: {{ .Release.Service | quote }}
         role: master
       {{- with .Values.master.podLabels }}
-      {{- toYaml . | nindent 8 }}
+      {{- toYaml . | indent 8 }}
       {{- end }}
       {{- with .Values.master.podAnnotations }}
       annotations: {{- toYaml . | nindent 8 }}
@@ -83,9 +83,9 @@
             - -cx
             - |
               {{- if .Values.persistence.enabled }}
-              mkdir -p {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
-              chmod 700 {{ .Values.persistence.mountPath }}/conf {{ .Values.persistence.mountPath }}/data
-              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | \
+              mkdir -p {{ .Values.persistence.mountPath }}/data {{- if (include "postgresql.mountConfigurationCM" .) }} {{ .Values.persistence.mountPath }}/conf {{- end }}
+              chmod 700 {{ .Values.persistence.mountPath }}/data {{- if (include "postgresql.mountConfigurationCM" .) }} {{ .Values.persistence.mountPath }}/conf {{- end }}
+              find {{ .Values.persistence.mountPath }} -mindepth 1 -maxdepth 1 {{- if not (include "postgresql.mountConfigurationCM" .) }} -not -name "conf" {{- end }} -not -name ".snapshot" -not -name "lost+found" | \
               {{- if eq ( toString ( .Values.volumePermissions.securityContext.runAsUser )) "auto" }}
                 xargs chown -R `id -u`:`id -G | cut -d " " -f2`
               {{- else }}
diff -ur a/values-production.yaml b/values-production.yaml
--- a/values-production.yaml	2020-04-23 09:55:01.000000000 +0200
+++ b/values-production.yaml	2020-04-27 11:41:58.000000000 +0200
@@ -15,7 +15,7 @@
 image:
   registry: docker.io
   repository: bitnami/postgresql
-  tag: 11.7.0-debian-10-r82
+  tag: 11.7.0-debian-10-r86
   ## Specify a imagePullPolicy
   ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
   ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
@@ -511,7 +511,7 @@
   image:
     registry: docker.io
     repository: bitnami/postgres-exporter
-    tag: 0.8.0-debian-10-r87
+    tag: 0.8.0-debian-10-r91
     pullPolicy: IfNotPresent
     ## Optionally specify an array of imagePullSecrets.
     ## Secrets must be manually created in the namespace.
diff -ur a/values.yaml b/values.yaml
--- a/values.yaml	2020-04-23 09:55:01.000000000 +0200
+++ b/values.yaml	2020-04-27 11:41:58.000000000 +0200
@@ -15,7 +15,7 @@
 image:
   registry: docker.io
   repository: bitnami/postgresql
-  tag: 11.7.0-debian-10-r82
+  tag: 11.7.0-debian-10-r86
   ## Specify a imagePullPolicy
   ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
   ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
@@ -517,7 +517,7 @@
   image:
     registry: docker.io
     repository: bitnami/postgres-exporter
-    tag: 0.8.0-debian-10-r87
+    tag: 0.8.0-debian-10-r91
     pullPolicy: IfNotPresent
     ## Optionally specify an array of imagePullSecrets.
     ## Secrets must be manually created in the namespace.
```

**Comment by middagj** [comment]
Already mentioned in: https://github.com/bitnami/charts/pull/2383#issuecomment-620097413

**Comment by dani8art** [comment]
Hi @middagj , It should be available in `bitnami/postgresql@8.9.3`, please could you give a try and  let us know if all is fine? thank you!

**Comment by carrodher** [comment]
Hi, you are totally right. After your PR was merged (bumping the chart version from `8.9.1` to `8.9.2`) another PR was merged without bump the chart version. Well, the chart version was bumped from `8.9.1` to `8.9.2` instead of `8.9.2` to `8.9.3` once your PR was merged.

This issue was not detected in the review process, nor GitHub merging the changes nor our internal CI/CD system. The result of that was that your changes were overwritten by the new PR. We are adding more checks to avoid this kind of corner cases in the future.

In order to solve the issue, we just released a new version (`8.9.3`) including your changes, you should to be able to see the changes by updating the helm repository and fetching the latest version:
```
▶ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈ Happy Helming!⎈

▶ helm fetch bitnami/postgresql

▶ tar xzf postgresql-8.9.3.tgz && cd postgresql

▶ ag '{{- with .Values.master.podLabels }}' -A 2
templates/statefulset.yaml
42:      {{- with .Values.master.podLabels }}
43-      {{- toYaml . | nindent 8 }}
44-      {{- end }}
```

We apologize for the inconvenience.

**Comment by middagj** [comment]
Thanks for the quick response. I thought it was something like that. Everything works again with version 8.9.3.
